README.md

```diff
@@ -2,4 +2,43 @@
 
 # Description
 
-Utilities for writing [sf](https://github.com/salesforcecli/cli) plugins.
+The @salesforce/sf-pluins-core provides utilities for writing [sf](https://github.com/salesforcecli/cli) plugins.
+
+## SfCommand Abstract Class
+
+The SfCommand abstract class extends @oclif/command class and adds useful extensions to ease the development of commands for use in the Salesforce Unified CLI.
+
+- SfCommand takes a generic type that defines the success JSON result
+- Enable the json flag support by default
+- Provides functions that help place success messages, warnings and errors into the correct location in JSON results
+- Enables additional help sections to the standard oclif command help output
+
+## Sf Hooks Utilities
+
+### SfHooks Interface
+
+Interface that defines the well known Unified CLI command hooks. See [@salesforce/plugin-deploy-retrieve-metadataA#src/hooks/deploy.ts](https://github.com/salesforcecli/plugin-deploy-retrieve-metadata/blob/main/src/hooks/deploy.ts) as an example implementation.
+
+### SfHook Class
+
+Class that provides a static method to run a well known hook.
+
+## Deployer Interface
```

Comment:
This could be confusing to developers who aren't familiar with what it means to "deploy" something and aren't familiar with how `sf` utilizes hooks

I think, instead, we should have a section that briefly describes the hook architecture and how the `Deployer` and `Deauthorizer` classes fit into that